test/unit/ready.js

```diff
@@ -67,4 +91,32 @@ QUnit.module( "ready" );
 			"Argument passed to fn in jQuery(document).ready( fn ) should be jQuery" );
 	} );
 
+	QUnit.test( "Promise.resolve(jQuery.ready)", function( assert ) {
+		assert.expect( 2 );
+		var done = jQuery.map( new Array( 2 ), function() { return assert.async(); } );
```

Comment:
This is a pattern we already use in several places. Regardless, we'll switch patterns when we update QUnit.